### PR TITLE
[WIP] Create .NET Core 2.0 slave based on slave-{nodejs,maven}

### PR DESCRIPTION
**Pull request by tmds**

### CODE REVIEW

File: slave-dotnet20/Dockerfile.rhel7 [hunk 1]
```diff
@@ -0,0 +1,30 @@
+FROM openshift3/jenkins-slave-base-rhel7
```
Comment (tmds) [inline slave-dotnet20/Dockerfile.rhel7]: The other slaves have `openshift/jenkins-slave-base-rhel7` as base, but I could not pull that, so changed it to `openshift3`

File: slave-dotnet20/Dockerfile.rhel7 [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+# Labels consumed by Red Hat build service
+LABEL com.redhat.component="jenkins-slave-dotnet20-rhel7-docker" \
```
Comment (tmds) [inline slave-dotnet20/Dockerfile.rhel7]: This slave specifically targets the 2.0 version of .NET Core. This matches with the scl packaging we have in mind. It means we will have a builder for each version of .NET Core (which are roughly released at 6M intervals).

File: slave-dotnet20/test/run [hunk 3]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash -e
```
Comment (tmds) [inline slave-dotnet20/test/run]: Who is calling the test script?

**Comment by tmds** [comment]
@bparees @jerboaa @omajid @aslicerh PTAL

File: slave-dotnet20/Dockerfile.rhel7 [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+ENV BASH_ENV=/usr/local/bin/scl_enable \
+    ENV=/usr/local/bin/scl_enable \
+    PROMPT_COMMAND=". /usr/local/bin/scl_enable"
+
+COPY contrib/bin/scl_enable /usr/local/bin/scl_enable
+
+# Install .NET Core
+RUN yum-config-manager --enable rhel-7-server-dotnet-rpms && \
+    yum-config-manager --disable epel >/dev/null || : && \
+    INSTALL_PKGS="rh-dotnet20" && \
```
Comment (tmds) [inline slave-dotnet20/Dockerfile.rhel7]: I assume we won't create a 1.x slave and start at 2.0 (due in a couple of months).

Comment (jerboaa) [inline slave-dotnet20/Dockerfile.rhel7]: FWIW, 1.x series still have some lifetime left. Perhaps those should be considered to be supported via jenkins slaves too? Some food for thought.

**Comment by tmds** [comment]
Q: the nodejs slave includes nodemon, why is it useful to have that in a slave?

File: slave-dotnet20/Dockerfile.rhel7 (hunk 1 above)
Comment (bparees) [inline slave-dotnet20/Dockerfile.rhel7]: that's an artifact of how we build internally vs what gets published not being quite aligned.  we'll need this to be openshift for our internal CI flow to work, for now.

File: slave-dotnet20/Dockerfile.rhel7 (hunk 2 above)
Comment (bparees) [inline slave-dotnet20/Dockerfile.rhel7]: sgtm.

File: slave-dotnet20/Dockerfile.rhel7 (hunk 4 above)
Comment (bparees) [inline slave-dotnet20/Dockerfile.rhel7]: i'm indifferent but maybe ask on this RFE bug whether 1.x is desirable:
https://bugzilla.redhat.com/show_bug.cgi?id=1451403

File: slave-dotnet20/test/run (hunk 3 above)
Comment (bparees) [inline slave-dotnet20/test/run]: it's called from the build.sh script when you run make test:
https://github.com/openshift/jenkins/blob/master/hack/build.sh#L66

which means we call it as part of our CI infrastructure.

**Comment by bparees** [comment]
@tmds thanks for doing this... please add your new slave to our makefile:
https://github.com/openshift/jenkins/blob/master/Makefile#L11

Also can you add a Centos Dockerfile version?

**Comment by bparees** [comment]
> Q: the nodejs slave includes nodemon, why is it useful to have that in a slave?

probably no good reason, we just grabbed all the same packages we were installing in the s2i nodejs image to ensure anything people were doing there, would work here too.  https://github.com/sclorg/s2i-nodejs-container/blob/master/4/Dockerfile#L39

**Comment by bparees** [comment]
related rfe: https://bugzilla.redhat.com/show_bug.cgi?id=1451403

**Comment by jerboaa** [comment]
> Also can you add a Centos Dockerfile version?

@bparees Note that the .NET collections are not yet available on CentOS. It'll depend on [this issue]( https://github.com/redhat-developer/s2i-dotnetcore/issues/36) being resolved before we can add this.

**Comment by bparees** [comment]
> @bparees Note that the .NET collections are not yet available on CentOS. It'll depend on this issue being resolved before we can add this.

any sense what the ETA looks like for that?

**Comment by omajid** [comment]
> any sense what the ETA looks like for that?

I think it depends on a few things:

- We need .NET Core 2.0 available for CentOS. This will probably happen soon after .NET Core 2.0 itself (or a pre-release version of that) is available for RHEL. I have reached out to the folks who handle this.
- We need to publish .NET Core 2.0 container images/`Dockerfile`s for CentOS.

**Comment by bparees** [comment]
>  This will probably happen soon after .NET Core 2.0 itself (or a pre-release version of that) is available for RHEL

you mean available for RHEL other than via software collections?

**Comment by omajid** [comment]
> you mean available for RHEL other than via software collections?

Err. No. Just available for RHEL, in general. NET Core 2.0 has not been released yet, though it's scheduled for summer.

**Comment by bparees** [comment]
> Err. No. Just available for RHEL, in general. NET Core 2.0 has not been released yet, though it's scheduled for summer.

then what exactly is being installed by the Dockerfile in this PR?
https://github.com/openshift/jenkins/pull/295/files#diff-ddf4e408efadd16916d19cedf16e0a5dR19

**Comment by omajid** [comment]
That's the expected name of .NET Core 2.0 package. It's just not available yet. I believe we will wait until the package is available before asking this PR to be merged in. 

I work with @jerboaa and @tmds so we will make sure to coordinate that.

**Comment by bparees** [comment]
> That's the expected name of .NET Core 2.0 package. It's just not available yet. 

oooooooh. :)

cool. sorry for jumping the gun, then.

**Comment by jerboaa** [comment]
@tmds We should port this for .NET 1.x for the time being. That's what is being asked for at the moment. Thoughts?

**Comment by tmds** [comment]
@jerboaa If we do this for 1.x, this will only work for `project.json`. Is that what is being asked for?

**Comment by jerboaa** [comment]
@tmds Yes. 1.x is sufficient for now. That's what we've been told.

**Comment by jerboaa** [comment]
I'm closing this PR since the image will be delivered via some other channel. A new PR will be opened that adds relevant config for that image.

**Pull request closed without merging**